### PR TITLE
feat: circulating supply helper function

### PR DESCRIPTION
**Pull request by jtfirek**
Hey @nud3l, am I on the right track for issue #94? I have created a function that will calculate the circulating supply. How would you like me to expose this function in the graphQL schema? Something like this maybe
```
import { GraphQLServer } from 'graphql-yoga';

const typeDefs = `
  type Query {
    getCirculatingSupply: BigInt!
  }
`;

const resolvers = {
  Query: {
    getCirculatingSupply: async () => {
      const circulatingSupply = await getCirculatingSupply();
      return circulatingSupply.toFixed(); // convert the Big value to a string
    },
  },
};

```

### CODE REVIEW

**Review by nud3l — CHANGES_REQUESTED** [review]
The on-chain code was intended as a reference. You will need to reimplement this using the on-chain events from the vesting schedules and the transfer events from the "special" pallets - like the `supply` and `escrow` pallets.

Squid should not make any on-chain calls as it will heavily slow down the processor. So those functions will need to be removed again.

As for the schema, I think something along the lines of the cumulative volumes would work:

```
cumulativeSupply {
  totalSupply: X
  circulatingSupply: Y
  totalVestingTokens: Z
  totalEscrowTokens: A
}
```

**Comment by gregdhill** [comment]
The circulating supply is calculated as `total_issuance - total_locked - total_reserved - system_account_supply` (@nud3l please correct me if I am wrong) so we can build this mostly from `orml_tokens` events:
- https://github.com/open-web3-stack/open-runtime-module-library/blob/0085382cf001704d084ac8e97b7bc3e4e8afb8a7/tokens/src/lib.rs#L281-L292
- https://github.com/open-web3-stack/open-runtime-module-library/blob/0085382cf001704d084ac8e97b7bc3e4e8afb8a7/tokens/src/lib.rs#L333-L357 

Basically I would initialize some `total_issuance` variable per currency (perhaps using [this event](https://github.com/open-web3-stack/open-runtime-module-library/blob/0085382cf001704d084ac8e97b7bc3e4e8afb8a7/tokens/src/lib.rs#L309-L313)) and track `total_locked` and `total_reserved` separately. The helper function could then read the balances of any system accounts (such as treasury) and then deduct those from `total_issuance` to get the circulating supply.

**Comment by nud3l** [comment]
For total_issuance, it's safe to set this to constant values for KINT and INTR each.

Agree with the orml-token event approach - we can just track the reserved (total_reserved) and frozen (total_locked) via these events and then deduct the system accounts as well.

For system accounts, it might be a bit more involved. The ones that I have in my mind are:
- supply pallet account
- vault annuity account
- escrow annuity account
- dex pool accounts (via the pool creating event)
- lending accounts
- treasury account

@gregdhill anything else we might need?

**Comment by gregdhill** [comment]
> For total_issuance, it's safe to set this to constant values for KINT and INTR each.

We should use this to set some initial variable and watch for changes since right now it is possible for the total issuance to increase.

@nud3l here is a list of all primary system accounts on Kintsugi: https://github.com/interlay/interbtc/blob/b2362d4a41c10bee0786cd48b592370cb4124bfb/parachain/runtime/kintsugi/src/lib.rs#L639-L658

There are also sub-accounts we will need to track dynamically for the following pallets:
- `DexGeneral`
- `DexStable`
- `Farming`
- `Loans`
